File: .tmp_py4013/batch_04_sources/2477_minimum_fuel_cost_to_report_to_the_capital.js
// LeetCode 2477 - Minimum Fuel Cost to Report to the Capital
// https://leetcode.com/problems/minimum-fuel-cost-to-report-to-the-capital/

/**
 * @param {number[][]} roads
 * @param {number} seats
 * @return {number}
 */
var minimumFuelCost = function(roads, seats) {
    const n = roads.length + 1;
    const g = Array.from({ length: n }, () => []);
    for (const [a, b] of roads) {
        g[a].push(b);
        g[b].push(a);
    }
    let ans = 0;
    const dfs = (u, p) => {
        let people = 1;
        for (const v of g[u]) {
            if (v !== p) people += dfs(v, u);
        }
        if (u !== 0) ans += Math.floor((people + seats - 1) / seats);
        return people;
    };
    dfs(0, -1);
    return ans;
};

File: .tmp_rs4013/rust_check_batch08/2477_minimum_fuel_cost_to_report_to_the_capital.rs
struct Solution;
// LeetCode 2477 - Minimum Fuel Cost to Report to the Capital
// https://leetcode.com/problems/minimum-fuel-cost-to-report-to-the-capital/

impl Solution {
    pub fn minimum_fuel_cost(roads: Vec<Vec<i32>>, seats: i32) -> i64 {
        let n = roads.len() + 1;
        let mut g = vec![Vec::new(); n];
        for r in &roads {
            let (a, b) = (r[0] as usize, r[1] as usize);
            g[a].push(b);
            g[b].push(a);
        }
        let mut ans = 0i64;
        fn dfs(u: usize, p: i32, seats: i32, g: &[Vec<usize>], ans: &mut i64) -> i32 {
            let mut people = 1;
            for &v in &g[u] {
                if v as i32 != p {
                    people += dfs(v, u as i32, seats, g, ans);
                }
            }
            if u != 0 {
                *ans += ((people + seats - 1) / seats) as i64;
            }
            people
        }
        dfs(0, -1, seats, &g, &mut ans);
        ans
    }
}

fn main() {}
